### ISSUE TITLE
Optimize computation graph

### ISSUE THREAD

**Issue by CemAlpturk** [CemAlpturk/Torchlet#3]
Don't include tensors whose gradients aren't tracked. Basically add `requires_grad` logic to `Tensor` class.
